src/withWebViewBridge.tsx

```diff
@@ -142,36 +162,46 @@ export const withWebViewBridge = function<WebViewType extends BaseWebViewType, W
         /* private */ _onNavigationStateChange(event: any) {
             //language=JavaScript
             this.webview.injectJavaScript(`
-                window.getReactNativeApi = (timeout = 1000) => new Proxy({}, {
-                    get(_, name) {
-                        return (arg) =>
-                            new Promise(function (resolve, reject) {
-                                const invocationId = btoa(Math.random()).substring(0, 12);
-                                window.addEventListener('message', function _listener({data}) {
-                                    if (data.invocationId === invocationId) {
-                                        window.removeEventListener('message', _listener, false);
-                                        if (data.type === 'functionResponse') {
-                                            resolve(data.data);
-                                        } else if (data.type === 'functionRejection') {
-                                            reject(data.data);
-                                        } else {
-                                            reject(new Error('unexpected response: ' + data.type));
-                                        }
+                if(window.getReactNativeApi === undefined) {
+                    window.getReactNativeApi = (timeout = 1000) => new Proxy({}, {
+                        get(_, name) {
+                            return (arg) =>
+                                new Promise(function (resolve, reject) {
+                                    try {
+                                        const invocationId = btoa(Math.random()).substring(0, 12);
+                                        window.addEventListener('message', function _listener({data}) {
+                                            if (data.invocationId === invocationId) { // if there is no data it will be an uncaught exception
+                                                window.removeEventListener('message', _listener, false);
+                                                try {
+                                                    if (data.type === 'functionResponse') {
+                                                        resolve(data.data);
+                                                    } else if (data.type === 'functionRejection') {
+                                                        reject(data.data);
```

Comment:
```suggestion
                                                        if(data.data !== undefined) {
                                                            reject(data.data);
                                                        } else {
                                                            reject(new Error(`data.data is undefined: ${data}`));
                                                        }
```

maybe something like this just in case?